Bottom: 117081856d7844d188c9986e48836e8bc449925f
Top:    d9e1fa658128fae0d78ba94130da2d9d0d37808c
Author: Jason Wang <jasowang@redhat.com>
Date:   2018-03-14 10:33:31 +0800

Refresh of tuntap-xdp_tx-can-use-native-0

---

diff --git a/drivers/net/tun.c b/drivers/net/tun.c
index 3023a93..bc18434 100644
--- a/drivers/net/tun.c
+++ b/drivers/net/tun.c
@@ -1670,7 +1670,7 @@ static struct sk_buff *tun_build_skb(struct tun_struct *tun,
 			preempt_enable();
 			return NULL;
 		case XDP_TX:
-			if (tun_xdp_xmit(tun->dev, &xdp))
+			if (tun_xdp_xmit(tun->dev, tun_xdp_to_ptr(&xdp)))
 				goto err_redirect;
 			tun_xdp_flush(tun->dev);
 			rcu_read_unlock();
